src/main/java/frc/team2412/robot/subsystems/ArmSubsystem.java

```diff
@@ -1,5 +1,155 @@
+
+		// Arm Positions
+
+		public static final double HIGH_NODE_ARM_ANGLE = 56.99;
+		public static final double MIDDLE_NODE_ARM_ANGLE = 92;
+		public static final double LOW_GRAB_ARM_ANGLE = 0;
+		public static final double RETRACT_ARM_ANGLE = 169;
+		public static final double SUBSTATION_ARM_ANGLE = 80;
+
+		// find values later
+		public static final double HIGH_NODE_WRIST_ANGLE = 0;
+		public static final double MIDDLE_NODE_WRIST_ANGLE = 0;
+		public static final double LOW_GRAB_WRIST_ANGLE = 0;
+		public static final double RETRACT_WRIST_ANGLE = 90;
+		public static final double SUBSTATION_WRIST_ANGLE = 0;
+
+		// LIKELY DONT NEED
+		// Apparently, we wanted the node position to be in ticks.
+		// We need an inches to ticks converter.
+		public static final double HIGH_NODE_CUBE_POS = 0;
+		public static final double HIGH_NODE_CONE_POS = 0;
+		public static final double MIDDLE_NODE_CUBE_POS = 0;
+		public static final double MIDDLE_NODE_CONE_POS = 0;
+		public static final double GRAB_LOW_POS = 0;
+		public static final double RETRACT_POS = 0;
+		public static final double SUBSTATION_POS = 0;
+
+		public static final double TICKS_TO_INCHES = 42 / 360; // Maybe wrong?
+
+		public static final int ARM_ANGLE_LIMIT = 20; // to find values later
+		public static final int WRIST_ANGLE_LIMIT = 20; // to find values later
+	}
+
+	// Hardware
+
+	private final CANSparkMax armMotor;
+	private final CANSparkMax wristMotor;
+
+	private final Encoder shoulderEncoder;
+	private final Encoder wristEncoder;
+
+	private final ProfiledPIDController armPID;
+	private final ProfiledPIDController wristPID;
+
+	// Constructor
+
+	public ArmSubsystem() {
+		armMotor = new CANSparkMax(20, MotorType.kBrushless);
+		wristMotor = new CANSparkMax(21, MotorType.kBrushless);
+		shoulderEncoder = new Encoder(SHOULDER_ENCODER_PORT_A, SHOULDER_ENCODER_PORT_B);
+		wristEncoder = new Encoder(WRIST_ENCODER_PORT_A, WRIST_ENCODER_PORT_B);
+
+		armPID = new ProfiledPIDController(K_P, K_I, K_D, ARM_CONSTRAINTS);
+		wristPID = new ProfiledPIDController(K_P, K_I, K_D, WRIST_CONSTRAINTS);
+
+		armMotor.setIdleMode(IdleMode.kBrake);
+		wristMotor.setIdleMode(IdleMode.kBrake);
+		armPID.setTolerance(ARM_POS_TOLERANCE, ARM_VELOCITY_TOLERANCE);
+		wristPID.setTolerance(WRIST_POS_TOLERANCE, WRIST_VELOCITY_TOLERANCE);
+
+		shoulderEncoder.reset();
+		wristEncoder.reset();
+	}
+
+	// Methods
+
+	// might not need?
+	// public double getVerticalArmPos() {
+	// 	return ((ARM_LENGTH * Math.sin(getShoulderAngle()))
+	// 			+ (VIRTUAL_BAR_LENGTH * Math.sin(getElbowAngle())));
+	// }
+
+	// public double getHorizontalArmPos() {
+	// 	return ((ARM_LENGTH * Math.cos(getShoulderAngle()))
+	// 			+ (VIRTUAL_BAR_LENGTH * Math.cos(getElbowAngle())));
+	// }
+
+	/** Stops Arm */
+	public void stopArm() {
+		armMotor.stopMotor();
+	}
+
+	public void stopWrist() {
+		wristMotor.stopMotor();
+	}
+
+	public void rotateArmTo(double targetAngle) {
+		armPID.setGoal(targetAngle);
```

Comment:
We may consider stopping motor if we are already at the targetAngle.

Reply:
I just instead did a checker so that the code wouldnt run if its already at the angle, the command itself should handle the stopping